src/userAgent.ts

```diff
@@ -0,0 +1,22 @@
+import { AxiosHeaders } from "axios";
+import { ExtendedAxiosRequestConfig } from "./NetworkProviderConfig";
+
+export function setUserAgent(userAgentPrefix: string, config: ExtendedAxiosRequestConfig | undefined) {
```

Comment:
In the current calls of this function, `config` is never `undefined`. Thus, we can simplify the function signature (and the first lines of the implementation / dropping the first check).

Reply:
done